_protected/app/system/core/assets/cron/24h/BannedCoreCron.php

```diff
@@ -0,0 +1,275 @@
+     * @var array
+     */
+    private $ipNew;
+
+    /**
+     * Contain existing blocked IP
+     *
+     * @var array
+     */
+    private $ipOld;
+
+    /**
+     * IP extracting regulat expression
+     *
+     * @var string
+     */
+    private $ipRegExp;
+
+    /**
+     * Contain the URL of the service we call to get banned IP
+     * Currently filled at instantiation staticly, will use config file later
+     *
+     * @var array
+     */
+    const SVC_URL = [
+        'https://www.blocklist.de/downloads/export-ips_all.txt',
+        'http://www.badips.com/get/list/ssh/2'
+    ];
+
+    public function __construct()
+    {
+        parent::__construct();
+        
+        /**
+         * Set valid IP regular expression using lazy mode (false)
+         *
+         * @var \PH7\BannedCoreCron $ipRegExp
+         */
+        $this->ipRegExp = self::regexpIP();
+        
+        /**
+         * Set the banned IP file name to the same file as used by PH7\Framework\Security\Ban\Ban
+         *
+         * @var \PH7\BannedCoreCron $bannedFileName
+         */
+        $this->bannedFileName = PH7_PATH_APP_CONFIG . Ban::DIR . Ban::IP_FILE;
+        $this->doProcess();
+    }
+
+    /**
+     * Get the job done !
+     */
+    protected function doProcess()
+    {
+        /**
+         * Process each web url we have in the $svcUrl array
+         */
+        foreach ($this->SVC_URL as $url) {
+            /**
+             * Each url we have for Web Service
+             *
+             * @var string $url
+             */
+            try {
+                /**
+                 * If we don't get true then we have an error
+                 */
+                if (! $this->callWebService($url)) {
+                    (new Logger())->msg('Error calling web service for banned IP url name :' . $url);
+                }
+            /**
+             * We catch exception so we can continue if one service fail
+             */
+            } catch (Exception $e) {
+                (new Logger())->msg('Error calling web service for banned IP url name :' . $url);
+            }
+        }
+        
+        /**
+         * Process the currently banned IP
+         */
+        $this->processExistingIP();
+        /**
+         * Merge both IPs and filter out doubles
+         */
+        $this->processIP();
+        
+        /**
+         * Wrie the new banned IP file
+         */
+        if (! $this->writeIP()) {
+            (new Logger())->msg('Error writting new banned IP file');
+        }
+    }
+
+    /**
+     * Call the web service with the given url and add received IP into $ipNew
+     *
+     * @param string $url
+     */
+    private function callWebService($url)
+    {
+        if (is_null($this->webClient)) {
+            $this->webClient = new \GuzzleHttp\Client();
+        }
+        
+        /**
+         * If we don't have a valid array to put address into, we create it.
+         */
+        if (! is_array($this->ipNew)) {
```

Comment:
You can refactoring this boolean method into a method like `isNewIp()`
I would always prefer a single readable method that says what it does instead of comments above conditions, etc.